src/black/__init__.py

```diff
@@ -1361,9 +1264,50 @@ def patch_click() -> None:
             module._verify_python_env = lambda: None  # type: ignore
 
 
+def shutdown(loop: "asyncio.AbstractEventLoop") -> None:
+    """Cancel all pending tasks on `loop`, wait for them, and close the loop."""
+    from black import concurrency
+
+    return concurrency.shutdown(loop)
+
+
+def cancel(tasks: Iterable["asyncio.Task[Any]"]) -> None:
+    """asyncio signal handler that cancels all `tasks` and reports to stderr."""
+    from black import concurrency
+
+    return concurrency.cancel(tasks)
+
+
+async def schedule_formatting(
+    sources: Set[Path],
+    fast: bool,
+    write_back: WriteBack,
+    mode: Mode,
+    report: "Report",
+    loop: "asyncio.AbstractEventLoop",
+    executor: "Executor",
+) -> None:
+    from black import reformat_many
+
+    return await reformat_many.schedule_formatting(
+        sources, fast, write_back, mode, report, loop, executor
+    )
+
+
+# Used to know whether uvloop should be installed.
+__BLACK_MAIN_CALLED__ = False
+
+
 def patched_main() -> None:
-    maybe_install_uvloop()
-    freeze_support()
+    global __BLACK_MAIN_CALLED__
+    __BLACK_MAIN_CALLED__ = True
+    for arg in sys.argv:
+        if "multiprocessing" in arg:
+            # Don't import multiprocessing unless really needed.
+            from multiprocessing import freeze_support
+
+            freeze_support()
+            break
```

Comment:
Is there documentation on that this is a valid way of using freeze_support on Windows? This feels exceedingly sketchy.